### PR TITLE
Write .htaccess also from CLI

### PR DESCRIPTION
**Pull request by LukasReschke**
The new updater as shipped with ownCloud 9.0.x invokes `occ` via `shell_exec`. This means that the `\OC::$CLI` code path is used when updating.

This removes the manual `.htaccess` modifications, effectively leading to the fact that URLs without index.php in it stop working. This also affects share URLs which could be considered a rather serious regression.
- User installs 9.0.0 via web
- User shares /s/1234
- User updates to 9.0.1 via ownCloud updater
- Link to /s/1234 is broken, /index.php/s/1234 works

@DeepDiver1975 What we discussed today.
@karlitschek Considering that this can be considered a regression caused by the new updater I propose a backport to 9.0.2

### CODE REVIEW

**Comment by karlitschek** [comment]
good catch. this should be backported 👍

**Comment by LukasReschke** [comment]
Backport at https://github.com/owncloud/core/pull/24136

File: lib/private/setup.php [hunk 1]
```diff
@@ -400,26 +400,34 @@ private function pathToHtaccess() {
...
 	 */
 	public static function updateHtaccess() {
-		// From CLI we don't know the defined web root. Thus we can't write any
-		// directives into the .htaccess file.
+		$config = \OC::$server->getConfig();
+
+		// For CLI read the value from overwrite.cli.url
 		if(\OC::$CLI) {
-			return;
+			$webRoot = $config->getSystemValue('overwrite.cli.url', '');
```
Comment (nickvergessen) [inline lib/private/setup.php]: `overwrite.cli.url` is the full URL btw, not only the webRoot.

Comment (LukasReschke) [inline lib/private/setup.php]: That's why it's overwritten 4 lines below using `parse_url`.

Comment (nickvergessen) [inline lib/private/setup.php]: :see_no_evil:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**